ftl/python/layer_builder.py

```diff
@@ -14,54 +14,227 @@
 """This package implements the Python package layer builder."""
 
 import datetime
+import logging
 import os
 import subprocess
-import logging
+import tempfile
 
 from ftl.common import ftl_util
 from ftl.common import single_layer_image
 from ftl.common import tar_to_dockerimage
 
 
 class PackageLayerBuilder(single_layer_image.CacheableLayerBuilder):
-    def __init__(self, ctx, descriptor_files, pkg_dir, dep_img_lyr):
+    def __init__(self, ctx=None, descriptor_files=None,
+                 pkg_dir=None, dep_img_lyr=None, cache=None):
         super(PackageLayerBuilder, self).__init__()
         self._ctx = ctx
         self._pkg_dir = pkg_dir
         self._descriptor_files = descriptor_files
         self._dep_img_lyr = dep_img_lyr
+        self._cache = cache
 
     def GetCacheKeyRaw(self):
         descriptor_contents = ftl_util.descriptor_parser(
             self._descriptor_files, self._ctx)
-        return "%s %s" % (descriptor_contents,
+        return '%s %s' % (descriptor_contents,
                           self._dep_img_lyr.GetCacheKeyRaw())
 
     def BuildLayer(self):
```

Comment:
might want to rename buildlayer here as it can also fetch a build layer from the cache

Reply:
ah yeah, good call

Reply:
actually going to do that in a separate PR, I want to also refactor the return values of that method on the builders